utils/misc.py

```diff
@@ -0,0 +1,16 @@
+def ordinal(n: int) -> str:
+    if n < 0:
+        raise ValueError('Negative ordinal - ordinal must be passed a non-negative integer.')
+
+    match n//10 % 10:  # for numbers trailing with 11, 12, or 13
+        case 1:
+            return f'{n}th'
```

Comment:
I'm not sure if I'm a fan of this tbh. I think it'd be clearer if it was just an if statement checking if n was 11, 12 or 13. I.e.
```python
if n in [11, 12, 13]:
    return f'{n}th'
```

I think this is clearer and neater than the current solution, especially as there is only one case in the match and its only ever for 11, 12 and 13.

Reply:
Actually I realize why you've done this now, because of 111, 112, 113, etc. I still think it'd be neater if you did the if statement but checking if `n%100` is either 11, 12 or 13, should be sufficient.

Reply:
Fair enough. Is it to make the specific distinction for "..11", "..12", and "..13" clearer in the code?
Something like this would suffice, right?
```py
    if n%100 in (11, 12, 13): return f'{n}th'
    if n%10 == 1: return f'{n}st'
    if n%10 == 2: return f'{n}nd'
    if n%10 == 3: return f'{n}rd'
    return f'{n}th'
```

Reply:
Yeah, I think the current method is a bit confusing for what its really doing, which is why I assume you have the comment in there. I thin something like that would also be perfectly fine. The match statement in the original was fine, so you can keep it if you want, up to you.